src/components/Constants/UseApiConstants.tsx

```diff
@@ -7,9 +9,22 @@ import {
 export const useApiConstants = ():
   | LoadConstantsQuery['constant']
   | undefined => {
-  const { data } = useLoadConstantsQuery({
+  const { data, refetch } = useLoadConstantsQuery({
     fetchPolicy: 'cache-first',
   });
 
+  const userSavedLanguage = data?.user.preferences?.locale;
+  const [localeStorageLanguage, setLocaleStorageLanguage] = useLocalStorage(
+    `user-language`,
+    '',
+  );
+  useEffect(() => {
```

Comment:
This is great! The only thing I'd suggest is making sure that this effect doesn't run when `userSavedLanguage` is undefined because it hasn't finished loading yet.